src/cls/rbd/cls_rbd.cc

```diff
@@ -2002,8 +2002,15 @@ int snapshot_remove(cls_method_context_t hctx, bufferlist *in, bufferlist *out)
     return r;
   }
 
+  cls_rbd_parent parent;
+  r = read_key(hctx, "parent", &parent);
+  if (r < 0 && r != -ENOENT) {
+    return r;
+  }
+  bool has_parent = (r != -ENOENT && parent.pool != -1);
```

Comment:
nit (could be just ignored): `r == 0` (or `>= 0`) looks a bit more appropriate to me here.